lib/riemann.js

```diff
@@ -9,8 +9,36 @@ var riemann = require("riemann");
  * Optional:
  * transport, flush, packet
  */
+
+
 function Riemann(startupTime, config, emitter) {
   this.config = config.riemann;
+  this.emitter = emitter;
+
+  if (this.config.transport && this.config.transport === "tcp") {
+    setInterval(this.reconnectIfNeeded.bind(this), this.config.reconnectInterval);
+  } else {
+    this.setupRiemannClient();
+  }
+}
+
+Riemann.prototype.reconnectIfNeeded = function() {
+    if (!this.client) {
+        if (this.config.debug) {
+            console.log('Connecting to Reimann');
+        }
+
+        this.setupRiemannClient();
+    } else if (this.client.tcp.socket.readyState == 'closed') {
+        if (this.config.debug) {
+            console.log('Reimann socket seems to be closed, reconnecting');
+        }
+
+        this.setupRiemannClient();
+    }
+}
+
+Riemann.prototype.setupRiemannClient = function() {
```

Comment:
Including `Riemann` in the function name seems a little unnecessary, rename to `setupClient`